### PR TITLE
Adds data model conformance tests for nulls and booleans

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

#88 

**Description of changes:**

Adds test cases for nulls and booleans as well as a few annotation test cases asserting that `true`, etc. cannot be used as annotations.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: conformance/data_model/annotations.ion [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//   * Annotations cannot be applied to NOP
+// * In all encodings
+//   * Any number of annotations is allowed (in any encoding)
+//   * Annotations can be applied to any type of value (all prior tests should be annotating a small integer)
+//
+// Less common scenarios:
+// * Add a test case with multiple values that are annotated to help ensure that there's
+//   no leaking annotation state between values
+
+(ion_1_x "in text, unquoted keywords cannot be used as an annotation"
```
Comment (zslayton) [inline conformance/data_model/annotations.ion:26]: Maybe add a `TODO` for the inverse of this? (i.e. Quoted keywords can be used as an annotation.)

File: conformance/data_model/null.ion [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+(ion_1_x "malformed nulls in text should signal an error"
+         (each (text "null.")
+               (text "null.1")
+               (text "null.ion")
+               (text "null.array")
+               (signals "invalid type")))
+
+(ion_1_x "in text, unquoted null cannot be used as an annotation"
+         (text "null::0")
+         (signals "cannot use unquoted keyword as an annotation"))
```
Comment (zslayton) [inline conformance/data_model/null.ion:29]: This is covered in a test case in `annotations.ion`.

File: conformance/data_model/null.ion [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+               (text "(null..)")
+               (text "(null.true)")
+               (text "(null.1)")
+               (text "(null.nan)")
+               (text "(null.+inf)")
+               (text "(null.foo)")
+               (text "(null.[])")
+               (text "(null.())")
+               (text "(null.{})")
+               (signals "invalid type")))
```
Comment (zslayton) [inline conformance/data_model/null.ion:43]: Nice! When we get to the tests for s-expressions, this will be one of a zillion corner cases to cover. 😵‍💫

File: conformance/data_model/null.ion [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+         (then (binary 0xEB 0x04) (denotes (Null timestamp)))
+         (then (binary 0xEB 0x05) (denotes (Null string   )))
+         (then (binary 0xEB 0x06) (denotes (Null symbol   )))
+         (then (binary 0xEB 0x07) (denotes (Null blob     )))
+         (then (binary 0xEB 0x08) (denotes (Null clob     )))
+         (then (binary 0xEB 0x09) (denotes (Null list     )))
+         (then (binary 0xEB 0x0A) (denotes (Null sexp     )))
+         (then (binary 0xEB 0x0B) (denotes (Null struct   ))))
+
+(ion_1_x "Null denotations"
```
Comment (zslayton) [inline conformance/data_model/null.ion:76]: Do I understand correctly that this is checking whether the `denotes` operator is working as intended?

**Review by zslayton — APPROVED** [review]

File: conformance/data_model/null.ion (hunk 4 above)
Comment (popematt) [reply conformance/data_model/null.ion:76]: I don't know. This was already part of the `nulls` file, so I kept it.

File: conformance/data_model/null.ion (hunk 2 above)
Comment (popematt) [reply conformance/data_model/null.ion:29]: Oops. Guess I forgot to remove it here when I generalized it in `annotations.ion`
